src/librustc/ty/structural_impls.rs

```diff
@@ -765,6 +769,36 @@ impl<'tcx> TypeFoldable<'tcx> for ty::RegionParameterDef<'tcx> {
     }
 }
 
+impl<'tcx> TypeFoldable<'tcx> for ty::Generics<'tcx> {
+    fn super_fold_with<'gcx: 'tcx, F: TypeFolder<'gcx, 'tcx>>(&self, folder: &mut F) -> Self {
+        ty::Generics {
+            parent: self.parent,
+            parent_regions: self.parent_regions,
```

Comment:
Unrelated to this PR: we should really implement folder on all types, so that this is just `fold_with()` no matter what.